packages/dds/map/src/test/mocha/directory.order.spec.ts

```diff
@@ -0,0 +1,509 @@
+			directory.createSubDirectory("c");
+
+			assertDirectoryIterationOrder(directory, ["b", "a", "c"]);
+		});
+
+		it("create nested subdirectories", () => {
+			directory.createSubDirectory("c").createSubDirectory("c-a");
+			directory.createSubDirectory("a").createSubDirectory("a-b");
+			directory.createSubDirectory("a").createSubDirectory("a-a");
+			directory.createSubDirectory("b");
+			directory.createSubDirectory("c").createSubDirectory("c-c");
+			directory.createSubDirectory("a").createSubDirectory("a-c");
+			directory.createSubDirectory("c").createSubDirectory("c-b");
+
+			assertDirectoryIterationOrder(directory, ["c", "a", "b"]);
+			assert.notEqual(directory.getWorkingDirectory("/a"), undefined);
+			assertDirectoryIterationOrder(directory.getWorkingDirectory("/a") as ISharedDirectory, [
+				"a-b",
+				"a-a",
+				"a-c",
+			]);
+			assert.notEqual(directory.getWorkingDirectory("/b"), undefined);
+			assertDirectoryIterationOrder(
+				directory.getWorkingDirectory("/b") as ISharedDirectory,
+				[],
+			);
+			assert.notEqual(directory.getWorkingDirectory("/c"), undefined);
+			assertDirectoryIterationOrder(directory.getWorkingDirectory("/c") as ISharedDirectory, [
+				"c-a",
+				"c-c",
+				"c-b",
+			]);
+		});
+
+		it("delete subdirectories", () => {
+			directory.createSubDirectory("c").createSubDirectory("c-a");
+			directory.createSubDirectory("a").createSubDirectory("a-b");
+			directory.createSubDirectory("a").createSubDirectory("a-a");
+			directory.createSubDirectory("b");
+			directory.createSubDirectory("c").createSubDirectory("c-c");
+			directory.createSubDirectory("a").createSubDirectory("a-c");
+			directory.createSubDirectory("c").createSubDirectory("c-b");
+
+			directory.deleteSubDirectory("a");
+			assertDirectoryIterationOrder(directory, ["c", "b"]);
+
+			directory.createSubDirectory("a");
+			assertDirectoryIterationOrder(directory, ["c", "b", "a"]);
+
+			directory.getWorkingDirectory("/c")?.createSubDirectory("c-d");
+			directory.getWorkingDirectory("/c")?.deleteSubDirectory("c-c");
+
+			assertDirectoryIterationOrder(directory.getWorkingDirectory("/c") as ISharedDirectory, [
+				"c-a",
+				"c-b",
+				"c-d",
+			]);
+		});
+	});
+
+	describe("Connected state", () => {
+		let containerRuntimeFactory: MockContainerRuntimeFactory;
+		let directory1: SharedDirectory;
+		let directory2: SharedDirectory;
+
+		beforeEach(async () => {
+			containerRuntimeFactory = new MockContainerRuntimeFactory();
+			// Create the first directory.
+			directory1 = createConnectedDirectory("directory1", containerRuntimeFactory);
+			// Create the second directory.
+			directory2 = createConnectedDirectory("directory2", containerRuntimeFactory);
+		});
+
+		it("Remote messages have no conflict with the local pending ops", () => {
+			directory1.createSubDirectory("a");
+			directory1.createSubDirectory("b").createSubDirectory("b-a");
+			directory2.createSubDirectory("c").createSubDirectory("c-b");
```

Comment:
What do the sub-subdirectories b-a and c-b add to this test?

Reply:
Have updated the test contents to leverage the child subdir